ui-v2/src/components/flows/table-count-header.tsx

```diff
@@ -0,0 +1,48 @@
+import { RowSelectionState } from "@tanstack/react-table";
+import { useMemo } from "react";
+import { Icon } from "../ui/icons";
+import { Typography } from "../ui/typography";
+
+export type TableCountHeaderProps = {
+	count: number;
+	rowSelectionState: RowSelectionState;
+	handleDeleteRows: () => void;
+};
+
+export const TableCountHeader = ({
+	count,
+	rowSelectionState,
+	handleDeleteRows,
+}: TableCountHeaderProps) => {
+	const selectedRows = useMemo(
+		() => Object.keys(rowSelectionState),
+		[rowSelectionState],
+	);
+
+	if (selectedRows.length > 0) {
+		return (
+			<Typography
+				variant="bodySmall"
+				className="text-muted-foreground flex items-center"
+				fontFamily="mono"
+			>
+				{selectedRows.length} selected{" "}
```

Comment:
nit: Rather than using a `{" "}` to create a space between the text and the icon, could we use flexbox and gap? I think the typography is already flexed, possibly just `gap-1` would be sufficient.